src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -172,10 +173,97 @@ public async Task GetAsync_SetCookieContainerMultipleCookies_CookiesSent()
+                        byte[] payloadBytes = Encoding.UTF8.GetBytes(payloadText);
+                        DelayedStream content = new DelayedStream(payloadBytes, delaySource.Task);
+                        request.Content = new StreamContent(content);
+                        requests.Add((client.SendAsync(request, HttpCompletionOption.ResponseContentRead), content));
+                    }
+
+                    HttpRequestMessage aboveLimitRequest = new HttpRequestMessage(HttpMethod.Post, Configuration.Http.Http2RemoteEchoServer) { Version = HttpVersion20.Value };
+                    aboveLimitRequest.Content = new StringContent($"{payloadText}-{maxActiveStreamsLimit + 1}");
+                    Task<HttpResponseMessage> aboveLimitResponseTask = client.SendAsync(aboveLimitRequest, HttpCompletionOption.ResponseContentRead);
+
+                    await aboveLimitResponseTask.TimeoutAfter(TestHelper.PassingTestTimeoutMilliseconds);
+                    await VerifyResponse(aboveLimitResponseTask, $"{payloadText}-{maxActiveStreamsLimit + 1}");
+
+                    delaySource.SetResult(true);
+
+                    await Task.WhenAll(requests.Select(r => r.task).ToArray()).TimeoutAfter(15000);
+
+                    foreach ((Task<HttpResponseMessage> task, DelayedStream stream) in requests)
+                    {
+                        Assert.True(task.IsCompleted);
+                        HttpResponseMessage response = task.Result;
+                        Assert.True(response.IsSuccessStatusCode);
+                        Assert.Equal(HttpVersion20.Value, response.Version);
+                        string responsePayload = await response.Content.ReadAsStringAsync().TimeoutAfter(TestHelper.PassingTestTimeoutMilliseconds);
+                        Assert.Contains(payloadText, responsePayload);
+                    }
+                }
+            }
+        }
+
+        private async Task VerifyResponse(Task<HttpResponseMessage> task, string payloadText)
+        {
+            Assert.True(task.IsCompleted);
+            HttpResponseMessage response = task.Result;
+            Assert.True(response.IsSuccessStatusCode);
+            Assert.Equal(HttpVersion20.Value, response.Version);
+            string responsePayload = await response.Content.ReadAsStringAsync().TimeoutAfter(TestHelper.PassingTestTimeoutMilliseconds);
+            Assert.Contains(payloadText, responsePayload);
+        }
+
         public static bool JsonMessageContainsKeyValue(string message, string key, string value)
         {
             string pattern = string.Format(@"""{0}"": ""{1}""", key, value);
             return message.Contains(pattern);
         }
+
+        private class DelayedStream : MemoryStream
+        {
+            private readonly Task _delayTask;
+            private readonly TaskCompletionSource<bool> _waitReadSource = new TaskCompletionSource<bool>();
+            private bool _completed;
```

Comment:
A better name might be `_delayed`... it's weird if there are outstanding reads still and `_completed` is true.